### ISSUE TITLE
not Accepting "Closed" as a string or as == "Closed"

### ISSUE THREAD

**Issue by William-Charles** [FreeCodeCamp/freeCodeCamp#12297]
Challenge [Exact Change](https://www.freecodecamp.com/challenges/exact-change#?solution=var%20changeDue%3B%0Avar%20drawerTotal%20%3D%200%3B%0A%0Afunction%20modularWhile(drawVal%2C%20curTar%2C%20curCoin)%7B%0A%20%20var%20counter%20%3D%200%3B%0A%20%20var%20arrBuild%20%3D%20%5B%5D%3B%0A%20%20%0A%20%20while(changeDue%20%3E%3D%20curCoin%20%26%26%20drawVal%20%3E%200)%7B%0A%20%20%20%20%0A%20%20%20%20if(counter%20%3D%3D%3D%200)%7B%0A%20%20%20%20%20%20arrBuild.push(%5BcurTar%2C%20curCoin%5D)%3B%0A%20%20%20%20%20%20counter%20%3D%201%3B%0A%20%20%20%20%20%20drawerTotal%20-%3D%20curCoin%3B%0A%20%20%20%20%20%20%0A%20%20%20%20%7D%0A%20%20%20%20else%7B%0A%20%20%20%20%20%20arrBuild%5B0%5D%5B1%5D%20%2B%3D%20curCoin%3B%0A%20%20%20%20%7D%0A%20%20%20%20drawVal%20-%3D%20curCoin%3B%0A%20%20%20%20changeDue%20-%3D%20curCoin%3B%0A%20%20%20%20drawerTotal%20-%3D%20curCoin%3B%0A%20%20%7D%0A%20%20%0A%20%20if(arrBuild.length%20%3E%200%20%26%26%20arrBuild%5B0%5D%5B0%5D%20%3D%3D%20%22PENNY%22)%7B%0A%20%20%20%20arrBuild%5B0%5D%5B1%5D%20%2B%3D%20curCoin%3B%0A%20%20%20%20changeDue%20-%3D%20curCoin%3B%0A%20%20%20%20drawerTotal%20-%3D%20curCoin%3B%0A%20%20%7D%0A%20%20return%20arrBuild%3B%0A%7D%0A%0A%0A%0A%0Afunction%20checkCashRegister(price%2C%20cash%2C%20cid)%20%7B%0A%20%20changeDue%20%3D%20cash%20-%20price%3B%0A%20%20var%20dollarVal%20%3D%20%5B0.01%2C%200.05%2C%200.1%2C%200.25%2C%201.00%2C%205.00%2C%2010.00%2C%2020.00%5D%3B%0A%20%20var%20change%20%3D%20%5B%5D%3B%0A%20%20%2F%2F%20Here%20is%20your%20change%2C%20ma'am.%0A%20%20%0A%20%20for(var%20i%20%3D%207%3B%20i%20%3E%20-1%3B%20i--)%7B%0A%20%20%20%20drawerTotal%20%2B%3D%20cid%5Bi%5D%5B1%5D%3B%0A%20%20%20%20var%20tempArr%20%3D%20modularWhile(cid%5Bi%5D%5B1%5D%2C%20cid%5Bi%5D%5B0%5D%2C%20dollarVal%5Bi%5D)%3B%0A%20%20%20%20if(tempArr.length%20%3E%200)%7B%0A%20%20%20%20%20%20var%20flat%20%3D%20tempArr.reduce(function(a%2C%20b)%20%7B%20%0A%20%20return%20a.concat(b)%3B%0A%7D%2C%20%5B%5D)%3B%0A%20%20%20%20%20%20change.push(flat)%3B%0A%20%20%20%20%7D%0A%20%20%7D%0A%20%20%0A%20%20%0A%20%20if(changeDue%20%3E%200.01)%7B%0A%20%20%20%20return%20%22Insufficient%20Funds%22%3B%0A%20%20%7D%0A%20%20else%20if(drawerTotal%20%3C%200.01)%7B%0A%20%20%20%20return%20'Closed'%3B%0A%20%20%7D%0A%20%20return%20change%3B%0A%7D%0A%0A%2F%2F%20Example%20cash-in-drawer%20array%3A%0A%2F%2F%20%5B%5B%22PENNY%22%2C%201.01%5D%2C%0A%2F%2F%20%5B%22NICKEL%22%2C%202.05%5D%2C%0A%2F%2F%20%5B%22DIME%22%2C%203.10%5D%2C%0A%2F%2F%20%5B%22QUARTER%22%2C%204.25%5D%2C%0A%2F%2F%20%5B%22ONE%22%2C%2090.00%5D%2C%0A%2F%2F%20%5B%22FIVE%22%2C%2055.00%5D%2C%0A%2F%2F%20%5B%22TEN%22%2C%2020.00%5D%2C%0A%2F%2F%20%5B%22TWENTY%22%2C%2060.00%5D%2C%0A%2F%2F%20%5B%22ONE%20HUNDRED%22%2C%20100.00%5D%5D%0A%0AcheckCashRegister(19.50%2C%2020.00%2C%20%5B%5B%22PENNY%22%2C%200.50%5D%2C%20%5B%22NICKEL%22%2C%200%5D%2C%20%5B%22DIME%22%2C%200%5D%2C%20%5B%22QUARTER%22%2C%200%5D%2C%20%5B%22ONE%22%2C%200%5D%2C%20%5B%22FIVE%22%2C%200%5D%2C%20%5B%22TEN%22%2C%200%5D%2C%20%5B%22TWENTY%22%2C%200%5D%2C%20%5B%22ONE%20HUNDRED%22%2C%200%5D%5D)%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

I am getting the correct answers according to the checks but it will not accept my solution when it returns "Closed"

My code:
```javascript
var changeDue;
var drawerTotal = 0;

function modularWhile(drawVal, curTar, curCoin){
  var counter = 0;
  var arrBuild = [];
  
  while(changeDue >= curCoin && drawVal > 0){
    
    if(counter === 0){
      arrBuild.push([curTar, curCoin]);
      counter = 1;
      drawerTotal -= curCoin;
      
    }
    else{
      arrBuild[0][1] += curCoin;
    }
    drawVal -= curCoin;
    changeDue -= curCoin;
    drawerTotal -= curCoin;
  }
  
  if(arrBuild.length > 0 && arrBuild[0][0] == "PENNY"){
    arrBuild[0][1] += curCoin;
    changeDue -= curCoin;
    drawerTotal -= curCoin;
  }
  return arrBuild;
}




function checkCashRegister(price, cash, cid) {
  changeDue = cash - price;
  var dollarVal = [0.01, 0.05, 0.1, 0.25, 1.00, 5.00, 10.00, 20.00];
  var change = [];
  // Here is your change, ma'am.
  
  for(var i = 7; i > -1; i--){
    drawerTotal += cid[i][1];
    var tempArr = modularWhile(cid[i][1], cid[i][0], dollarVal[i]);
    if(tempArr.length > 0){
      var flat = tempArr.reduce(function(a, b) { 
  return a.concat(b);
}, []);
      change.push(flat);
    }
  }
  
  
  if(changeDue > 0.01){
    return "Insufficient Funds";
  }
  else if(drawerTotal < 0.01){
    return "Closed";
  }
  return change;
}

// Example cash-in-drawer array:
// [["PENNY", 1.01],
// ["NICKEL", 2.05],
// ["DIME", 3.10],
// ["QUARTER", 4.25],
// ["ONE", 90.00],
// ["FIVE", 55.00],
// ["TEN", 20.00],
// ["TWENTY", 60.00],
// ["ONE HUNDRED", 100.00]]

checkCashRegister(19.50, 20.00, [["PENNY", 0.50], ["NICKEL", 0], ["DIME", 0], ["QUARTER", 0], ["ONE", 0], ["FIVE", 0], ["TEN", 0], ["TWENTY", 0], ["ONE HUNDRED", 0]]);

```

**Comment by dhcodes** [comment]
Something does seem to be amiss with this.

**Comment by William-Charles** [comment]
Whoa! Thank you!

**Comment by systimotic** [comment]
Nice work on figuring this out @MiloATH! 👍 

Global scope is something we've talked about before. I do not remember where, or what was decided. Do you know @dhcodes?

If we want to fix this, we should probably open a separate issue for discussion on how we handle global scope in general.

**Comment by raisedadead** [comment]
@dhcodes @systimotic are we still blocked on this?

**Comment by systimotic** [comment]
I found this issue https://github.com/FreeCodeCamp/freeCodeCamp/issues/9199 on global variables. I'm closing this on favour of that.
